### PR TITLE
Complete TODO comment in DefaultEntitiesCatalog.ts

### PR DESCRIPTION
**Pull request by schultzp2020**
## Hey, I just made a Pull Request!

Relations, as returned through the `relations` field of entities in the catalog REST API responses, no longer have the `target` property.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by freben — COMMENTED** [review]
Thanks. Letting this one simmer a bit

File: .changeset/wise-planes-listen.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+---
+'@backstage/plugin-catalog-backend': major
```
[the remark this answers is not in the document]
Comment (schultzp2020) [reply .changeset/wise-planes-listen.md:2]: @freben where do I ping `@backstage/catalog-maintainers`? I don't see a GitHub group I can ping.

[the remark this answers is not in the document]
Comment (freben) [reply .changeset/wise-planes-listen.md:2]: Sorry my comment wasn't clear. I didn't mean "ping" as an imperative, I pinged them by mentioning them :)

**Comment by freben** [comment]
@schultzp2020 Hi again! A suggestion to move this forward - how would you feel about:

- Leaving `/entities` completely unchanged
- Adding a new `/v2/entities` that does have the new behaviour
- Changing the catalog-client package to use this new endpoint (transparently to the user)
- Moving the TODO comment to be on the old `/entities` instead, saying clearly that this endpoint is deprecated and slated for removal when the next major release of the catalog is made

This would mean making a `minor` level change both to the catalog backend and the catalog client.

Would you be up for trying to make that change? 🙏

**Comment by schultzp2020** [comment]
@freben can you point me to the correct files for each step? I don't know where `/entities` is located or being used. Lastly, are you suggesting to create a new `DefaultEntitiesCatalog` class or should there be a parameter that turns the deprecation on/off?

**Pull request closed without merging**